### ISSUE TITLE
Windows Defender Flagged as PUS (Potentially Unwanted Software)

### ISSUE THREAD

**Issue by jimmydorry** [Syncplay/syncplay#458]
**Describe the bug**
Windows Defender has just flagged the syncplay server executable as PUS (Potentially Unwanted Software).

The learn more links to here: https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?name=Program%3aWin32%2fUwamson.A!ml&threatid=250070

**To Reproduce**
1. Triggered automatically when Windows Defender runs

**Expected behavior**
Syncplay executables should not trigger Windows Defender. This issue appears to be related to issue #456 

**Screenshots**
![image](https://user-images.githubusercontent.com/5135141/130734257-937fd826-315c-4b68-8047-efba8bd78bc3.png)

**Version and platform:**
 - OS: Windows 10 Pro 19041.508 (Windows Feature Experience Pack 120.2212.31.0)
 - Syncplay v 1.6.9 release 99 (Python 3.8.10 - PySide2 5.15.2 - Qt 5.15.2)

**Additional context**
I'm not aware of the process of getting this unflagged, but I imagine this is a shared database across many OS and Browser and Anti-Virus vendors.

**Comment by jimmydorry** [comment]
https://www.microsoft.com/en-us/wdsi/filesubmission 

I've submitted a case via this link saying it's a false positive. A project maintainer may want to do this in a more official capacity.

**Comment by jimmydorry** [comment]
As follow up, just to dot my I's and cross my T's, I attempted downloading `Syncplay-1.6.9-Setup.exe` straight from github and it's getting flagged in Chrome now too.

![image](https://user-images.githubusercontent.com/5135141/130735839-ea6c9fe4-4b7a-40f2-b3b8-b32e52e6c64e.png)

**Comment by jimmydorry** [comment]
Looks like this might be solved (for Windows Defender at least)

![image](https://user-images.githubusercontent.com/5135141/130796440-76a2d012-7dd8-4ccf-a3a9-4d3fe8a41645.png)

        Analyst comments:
        
        We have removed the detection. Please follow the steps below to clear cached detection and obtain the latest malware definitions.
        
        1. Open command prompt as administrator and change directory to c:\Program Files\Windows Defender
        2. Run “MpCmdRun.exe -removedefinitions -dynamicsignatures”
        3. Run "MpCmdRun.exe -SignatureUpdate"
        
        Alternatively, the latest definition is available for download here: https://www.microsoft.com/en-us/wdsi/definitions
        
        Thank you for contacting Microsoft.
        
        
        Click here for more information

**Comment by jimmydorry** [comment]
What gave the warning? Windows Defender? Chrome? Firefox?

So far, every list maintainer has said this was a false positive and removed it... but there are just so many lists. I haven't had any luch getting it off the Chrome list.

**Comment by kimpurcell** [comment]
Chrome, Windows Defender, even most vendors on VirusTotal flags it.

Not safe. Don't trust open source, think there's no trojan cuz source code is open? Think again.

**Comment by jimmydorry** [comment]
You may need to scroll up. 

The Windows Defender team responded to the ticket and removed the false positive from their latest database. 

VirusTotal shows negative when uploading the binary. https://www.virustotal.com/gui/url/0c834cdd0d49b5adc4a2a4cf9ed07b9ebb95a72c22d31c9a0e7dffe38c7729a5/details

**Comment by Dangerin23** [comment]
I received the warning on Windows Defender as I use a 3rd party downloader and not chrome downloader. So maybe it would have popped up on chrome as well. And this was yesterday.

**Comment by ImportTaste** [comment]
@Et0h In the meantime, I'd like to recommend that the syncplay server for Windows be packaged separately.

**Comment by kimpurcell** [comment]
> You may need to scroll up.
> 
> The Windows Defender team responded to the ticket and removed the false positive from their latest database.
> 
> VirusTotal shows negative when uploading the binary. https://www.virustotal.com/gui/url/0c834cdd0d49b5adc4a2a4cf9ed07b9ebb95a72c22d31c9a0e7dffe38c7729a5/details

That’s not the same binary as the one uploaded on the website. Pretty deceptive if you ask me. 

PSA: Running a malicious .exe is all it takes to get REKT. Hackers will have total control of your machine and they will cause maximum financial damage not only to you, but your family and friends as well. Don’t risk it.

**Comment by Et0h** [comment]
I just double checked and the installer and portable zip which got an all clear at https://www.virustotal.com/gui/url/0c834cdd0d49b5adc4a2a4cf9ed07b9ebb95a72c22d31c9a0e7dffe38c7729a5/details and https://www.virustotal.com/gui/url/00fde2631169ea88ccbaf0e7791fa362c63554a2ca08fd9d74f8f5891da74b4c/detection have the same SHA-256 checksums as the files available from the website (i.e. 5849d909cd786a07ffc88d68400d006303908c5a2469388b00439591bfe68570 and 80a5c8f1facb6fe6c5d9351ae1e8ff881727d637e0a316f7c5ab50796c6e3f5a respectively). I also checked with Windows Defender and it didn't find any threats, which is good.

However, it appears that some anti-virus on VirusTotal don't seem to handle .zip files when, and so if syncplayServer.exe or syncplay.exe is scanned on its own it sends it to some where I have not previously sent false positive reports. 

As you may be aware, false positives with third-party is quite common. Because Syncpaly cannot afford for its code to be signed it is at the mercy of anti-virus developers who don't really seem to care about how many false positives they have or how difficult it is to report them. Submitting false positive reports is time-consuming, so thanks to everyone who has submitted a false positive report.

I've now submitted false positive reports for all the remaining false positives making use of the list at https://github.com/yaronelh/False-Positive-Center but they can take a while to be processed. Options include waiting for them to be processed (and updating your virus defintions), trusting the Syncplay developers who have been releasing this free software for the best part of a decade, or run the open source code via Python which allows for full control over what is wrong.

I've never tried running Syncplay on Windows with sandboxing software, but I know that on some systems there are problems if the media player it executes is not part of the same sandbox. If Sandboxie does work then that would be interesting to know.

**Comment by kimpurcell** [comment]
That's not the same binary as the one on the website.
Download this: https://github.com/Syncplay/syncplay/releases/download/v1.6.9/Syncplay-1.6.9-Setup.exe

Results with over 12 positives here: https://www.virustotal.com/gui/file/5849d909cd786a07ffc88d68400d006303908c5a2469388b00439591bfe68570/detection/f-5849d909cd786a07ffc88d68400d006303908c5a2469388b00439591bfe68570-1630277057

**Comment by jimmydorry** [comment]
Thanks @Et0h The only major vendor I don't see in the list is Google / the Chrome Browser. I tried their Safe Browsing False Positive form, but that doesn't appear to have done anything after 4 days.

**Comment by Et0h** [comment]
> Thanks @Et0h The only major vendor I don't see in the list is Google / the Chrome Browser. I tried their Safe Browsing False Positive form, but that doesn't appear to have done anything after 4 days.

https://developers.google.com/safe-browsing/v4/reporting directed me to https://www.stopbadware.org/home/reviewinfo which directed me to https://safebrowsing.google.com/safebrowsing/report_error/?hl=en&visit_id=637658759659559232-2016553642&help_center_redirect=true&rd=1 where I reported the false positive. I hope that does the trick!

**Comment by ImportTaste** [comment]
> I just double checked and the installer and portable zip which got an all clear at https://www.virustotal.com/gui/url/0c834cdd0d49b5adc4a2a4cf9ed07b9ebb95a72c22d31c9a0e7dffe38c7729a5/details and https://www.virustotal.com/gui/url/00fde2631169ea88ccbaf0e7791fa362c63554a2ca08fd9d74f8f5891da74b4c/detection have the same SHA-256 checksums as the files available from the website (i.e. 5849d909cd786a07ffc88d68400d006303908c5a2469388b00439591bfe68570 and 80a5c8f1facb6fe6c5d9351ae1e8ff881727d637e0a316f7c5ab50796c6e3f5a respectively). I also checked with Windows Defender and it didn't find any threats, which is good.
> 
> However, it appears that some anti-virus on VirusTotal don't seem to handle .zip files when, and so if syncplayServer.exe or syncplay.exe is scanned on its own it sends it to some where I have not previously sent false positive reports.
> 
> As you may be aware, false positives with third-party is quite common. Because Syncpaly cannot afford for its code to be signed it is at the mercy of anti-virus developers who don't really seem to care about how many false positives they have or how difficult it is to report them. Submitting false positive reports is time-consuming, so thanks to everyone who has submitted a false positive report.
> 
> I've now submitted false positive reports for all the remaining false positives making use of the list at https://github.com/yaronelh/False-Positive-Center but they can take a while to be processed. Options include waiting for them to be processed (and updating your virus defintions), trusting the Syncplay developers who have been releasing this free software for the best part of a decade, or run the open source code via Python which allows for full control over what is wrong.
> 
> I've never tried running Syncplay on Windows with sandboxing software, but I know that on some systems there are problems if the media player it executes is not part of the same sandbox. If Sandboxie does work then that would be interesting to know.

I shared the portable zip of 1.6.9 just the other day with someone and they weren't able to download it because they kept getting it blocked by their antivirus. It's not an issue of us trusting you, it's an issue of having to convince computer-illiterate friends that it's actually safe and to turn off their antivirus for a few minutes.

**Comment by kimpurcell** [comment]
Y'all trying really hard to shill unsuspecting users. This software is NOT safe, being flagged by multiple vendors as having a trojan. Also, all of you posting the clean binaries are not addressing the fact that your checksums do NOT match the binaries on the website. 

Don't say I didn't warn you when these users hack your machines and steal your money!

**Comment by jimmydorry** [comment]
You should tone down your replies by several notches. The checksum on the site matches the checksum of my local copy that I downloaded from the Github... All of your replies have just been noise so far @kimpurcell 

![image](https://user-images.githubusercontent.com/5135141/131280001-5c9dda30-f8b8-4668-a7f0-9cea00f2aa38.png)

![image](https://user-images.githubusercontent.com/5135141/131280016-74f96f58-28d5-47bf-8711-1c1f7e0563bd.png)

**Comment by kimpurcell** [comment]
stop trying to lie, here's a screenshot proof that it's a bad binary:

https://www.virustotal.com/gui/file/5849d909cd786a07ffc88d68400d006303908c5a2469388b00439591bfe68570/detection/f-5849d909cd786a07ffc88d68400d006303908c5a2469388b00439591bfe68570-1630277057

![bad](https://user-images.githubusercontent.com/17389179/131280856-bf8b1fd8-bc3d-4ec4-be58-ee0e23b7bbf4.png)

**Comment by daniel-123** [comment]
@kimpurcell please keep it civil and consider this message a warning.

You can see our entire process of creating binaries and installer - it is using github actions [configured as follows](https://github.com/Syncplay/syncplay/blob/master/.github/workflows/build.yml). You can recreate the same binaries from scratch if you have the same environment or run those actions in your own forked repository. They will also get very similar flags in virustotal.

Problem with false positives being reported by antivirus software is nothing new and it's depressingly common occurrence for all niche software (it even affects historical releases of Syncplay to various degree). Besides dutifully flagging all false positives and waiting for antivirus software vendors to fix their detection there are no easy ways to fix it. The "proper" way is establishing a process of cryptographically signing the binaries, but logistics of that are far from trivial.

**Comment by Et0h** [comment]
For now I have changed the Windows download links on the website to https://github.com/Syncplay/syncplay/releases/download/v1.6.9/Syncplay-Client-1.6.9.zip and https://github.com/Syncplay/syncplay/releases/download/v1.6.9/Syncplay-Portable-Client-1.6.9.zip (which are just .zips of the files excluding syncplayServer.exe) as that doesn't seem to trigger Google Chrome.
 
In other news, I have had some confirmations from some anti-virus companies that they have flagged Syncplay as a false positive:
**VBA32:**
![image](https://user-images.githubusercontent.com/2389316/131558844-116f24ff-479e-4a68-bc68-e91cb01e27bd.png)
**ESTsecurity**
![image](https://user-images.githubusercontent.com/2389316/131558936-f7541d1b-9648-471a-b72a-390f2a01655e.png)
**G DATA**
![image](https://user-images.githubusercontent.com/2389316/131559064-b22c3b23-cc51-4348-a4c7-43942f063ca7.png)
**eScan** 
![image](https://user-images.githubusercontent.com/2389316/131559410-f21816e1-2624-487b-81d0-0db2d68c328d.png)

**Comment by jimmydorry** [comment]
Thanks @Et0h . Long term, as ImportTaste recommended above, you may want to consider bundling SyncplayServer separately for the windows binaries.

**Comment by Et0h** [comment]
> Thanks @Et0h . Long term, as ImportTaste recommended above, you may want to consider bundling SyncplayServer separately for the windows binaries.

Yes, given that most users don't need the server then it seems like a reasonable way of reducing the risk and impact of false positives going forwards.
